packages/openneuro-app/src/client.jsx

```diff
@@ -22,8 +22,28 @@ ReactDOM.render(
     <ApolloProvider
       client={createClient(`${config.url}/crn/graphql`, {
         clientVersion: version,
-        // @ts-expect-error
-        cache: new InMemoryCache().restore(JSON.parse(window.__APOLLO_STATE__)),
+        cache: new InMemoryCache({
+          typePolicies: {
+            DatasetConnection: {
+              fields: {
+                edges: {
+                  keyArgs: ['query', 'datasetType', 'datasetStatus', 'sortBy'],
+                  merge: (existing = [], incoming) => {
+                    existing = existing.filter(x => x)
+                    const existingIds = existing.map(({ node }) => node.__ref)
+                    incoming = incoming
+                      .filter(edge =>
+                        existingIds.includes(edge?.node.__ref) ? null : edge,
+                      )
+                      .filter(x => x)
```

Comment:
The fact this is required means this is still not correctly using keying the cached objects.

Reply:
Thanks for your changes here!